UnitTests/UnitTests.HighPerformance.Shared/Extensions/Test_ArrayExtensions.3D.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Microsoft.Toolkit.HighPerformance.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace UnitTests.HighPerformance.Extensions
+{
+    public partial class Test_ArrayExtensions
+    {
+        [TestCategory("ArrayExtensions")]
+        [TestMethod]
+        public void Test_ArrayExtensions_3D_DangerousGetReference_Int()
```

Comment:
It may be helpful for us to have brief descriptions of these tests to describe what the scenario is, what they're doing, and what the expected outcome is? It's been pretty hard to parse the raw code itself in these complex in-depth scenarios, so a bit of prose for context may help us better review these in the future?

Reply:
Right! I documented all the APIs but didn't think about also adding comments to the tests 😅
Added some for these APIs (and variants for 1D and 2D arrays too) in 0e8a2db02bde872c42f38f4d2e8fa259aca6bc74.
Will definitely keep in mind to document unit tests more in the future!